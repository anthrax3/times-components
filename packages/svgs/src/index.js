export { default as Svg } from "./svg";
export { default as G } from "./g";
export { default as Path } from "./path";
