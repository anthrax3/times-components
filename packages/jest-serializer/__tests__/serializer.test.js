import {View, Text, StyleSheet} from "react-native";
import {Path} from "svgs";
import React from "react";
import renderer from "react-test-renderer";
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe("jest serializer", () => {
  it("removes rnw-classnames", () => {
    const component = <View className="rn-prop-hash1 main rn-prop-hash2 main2" />
    const tree = renderer.create(
      component
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("adds styles", () => {
    const {colored, padded} = StyleSheet.create({
      colored: {
        color: "red",
      },
      padded: {
        padding: 1
      }
    });

    const component = <Text style={[colored, padded]} />
    const tree = renderer.create(component);

    expect(tree).toMatchSnapshot();
  });

  it("removes paths d props", () => {
    const component = <Path d="1 2 3 4 5 6"/>
    const tree = renderer.create(component);
    expect(tree).toMatchSnapshot();
  });

  it("transforms render props", () => {
    const prop = <Path d="1 2 3 4 5 6"/>
    const Nested = () => <Text> </Text>
    const PropRenderer = ({prop}) => <View><Nested prop={prop} /></View>;
    const tree = shallow(<PropRenderer prop={prop} />);
    expect(tree).toMatchSnapshot();
  });
});
