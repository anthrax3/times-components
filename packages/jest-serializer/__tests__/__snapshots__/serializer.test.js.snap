// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest serializer should not remove styles 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "red",
      "padding": 1,
    }
  }
/>
`;

exports[`Jest serializer should remove d-prop from path 1`] = `<path />`;

exports[`Jest serializer should remove rnw-classnames 1`] = `
<View
  className="main main2"
/>
`;

exports[`Jest serializer should transform render props 1`] = `
<View>
  <Nested
    prop={<Path />}
  />
</View>
`;
